File: aws-utils/awsip.py
#!/usr/bin/env python3
import json
import requests
import ipaddress
import sys

aws_ip_url = 'https://ip-ranges.amazonaws.com/ip-ranges.json'

try:
    ip = sys.argv[1]
except IndexError:
    raise SystemExit(f"Usage: {sys.argv[0]} <ip address to check>")

try:
    ip_ranges = requests.get(aws_ip_url, timeout=30).json()['prefixes']
except requests.exceptions.HTTPError as errh:
    raise SystemExit(errh)
except requests.exceptions.ConnectionError as errc:
    raise SystemExit(errc)
except requests.exceptions.Timeout as errt:
    raise SystemExit(errt)
except requests.exceptions.RequestException as err:
    raise SystemExit(err)


for net in ip_ranges:
    if ipaddress.IPv4Interface(ip) in ipaddress.IPv4Network(net["ip_prefix"]):
        print(json.dumps(net, indent=1))

File: aws-utils/awsip.rb
#!/usr/bin/env ruby

# Query the AWS IP range list, filter for the given IP/subnet, and spit
# the results back as JSON

require 'json'
require 'net/http'
require 'ipaddr'

AWS_IP_URI = URI('https://ip-ranges.amazonaws.com/ip-ranges.json')

req_ip = IPAddr.new(ARGV.shift) rescue abort("Usage: #{$0} ip[/prefix]")

aws_ips = nil

Net::HTTP.start(AWS_IP_URI.host, AWS_IP_URI.port,
  :use_ssl => AWS_IP_URI.scheme == 'https') do |http|
  request = Net::HTTP::Get.new AWS_IP_URI
  response = http.request request # Net::HTTPResponse object
  aws_ips = JSON.parse(response.body)
end

filtered = []

aws_ips['prefixes'].each do |pfx|
  pfx_ip = IPAddr.new(pfx['ip_prefix'])
  filtered << pfx if pfx_ip.include?(req_ip)
end

puts JSON.pretty_generate(filtered)
